src/clj/game/cards/programs.clj

```diff
@@ -850,7 +850,8 @@
             (if (empty? (filter program? cards))
               {:async true
                :msg "shuffle the stack"
-               :effect (effect (shuffle! :deck))}
+               :effect (req (shuffle! state side :deck)
+                            (effect-completed state side eid))}
```

Comment:
Because the ability only called system-msg, it doesn't need to be async, so remove this line and the :async true a couple lines above.